### PR TITLE
raft: improve test coverage

### PR DESCRIPTION
**Pull request by hhkbp2**
Related to #966

### CODE REVIEW

**Comment by hhkbp2** [comment]
Hold until https://github.com/coreos/etcd/pull/6541 is merged.

**Comment by hhkbp2** [comment]
https://github.com/coreos/etcd/pull/6541 merged.
PTAL @ngaut @siddontang @BusyJay

File: src/raft/raft.rs [hunk 1]
```diff
@@ -971,7 +971,7 @@ impl<T: Storage> Raft<T> {
                 let rss = self.read_only.advance(m);
                 for rs in rss {
                     let mut req = rs.req;
-                    if req.get_from() == INVALID_ID || m.get_from() == self.id {
+                    if req.get_from() == INVALID_ID || req.get_from() == self.id {
```
Comment (siddontang) [inline src/raft/raft.rs]: Any test to cover here?

Comment (hhkbp2) [inline src/raft/raft.rs]: The added test cases cover this bug.

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**